### PR TITLE
[SK-20] Configure django rest framework and drf-yasg.

### PR DESCRIPTION
**Pull request by MarcinMistarz**
Configuration of django rest framework and drf-yasg.

### CODE REVIEW

File: backend/sky_lectures/urls.py [hunk 1]
```diff
@@ -13,9 +13,30 @@
...
     2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
 """
+from django.conf.urls import url
 from django.contrib import admin
-from django.urls import path
+from django.urls import path, include
+from rest_framework import permissions, routers, serializers, viewsets
+from drf_yasg.views import get_schema_view
+from drf_yasg import openapi
+from django.contrib.auth.models import User
```
Comment (fiedosiukr) [inline backend/sky_lectures/urls.py]: Divide imports into sections separated by a blank line:
1. Standard imports
2. 3rd party imports
3. Application imports

File: backend/sky_lectures/urls.py [hunk 2]
```diff
@@ -13,9 +13,30 @@
...
+
+schema_view = get_schema_view(
+   openapi.Info(
+      title="Snippets API",
+      default_version='v1',
+      description="Test description",
+      terms_of_service="https://www.google.com/policies/terms/",
+      contact=openapi.Contact(email="contact@snippets.local"),
+      license=openapi.License(name="BSD License"),
```
Comment (fiedosiukr) [inline backend/sky_lectures/urls.py]: I guessed you forgot to change title, etc.

File: backend/sky_lectures/urls.py [hunk 3]
```diff
@@ -13,9 +13,30 @@
...
+   ),
+   public=True,
+   permission_classes=[permissions.AllowAny],
+)
 
 urlpatterns = [
     path('admin/', admin.site.urls),
+    url(r'^swagger(?P<format>\.json|\.yaml)$', schema_view.without_ui(cache_timeout=0), name='schema-json'),
+    url(r'^swagger/$', schema_view.with_ui('swagger', cache_timeout=0), name='schema-swagger-ui'),
+    url(r'^redoc/$', schema_view.with_ui('redoc', cache_timeout=0), name='schema-redoc'),
```
Comment (fiedosiukr) [inline backend/sky_lectures/urls.py]: **DO NOT USE URL!**
Use path (sometimes you may need to use repath).

Please read the documentation. You only need to provide one view. There is no need to provide three different ones.

Swagger should be only accessible during the development.

File: backend/sky_lectures/urls.py [hunk 4]
```diff
@@ -13,9 +13,30 @@
...
+      title="Snippets API",
+      default_version='v1',
+      description="Test description",
+      terms_of_service="https://www.google.com/policies/terms/",
+      contact=openapi.Contact(email="contact@snippets.local"),
+      license=openapi.License(name="BSD License"),
+   ),
+   public=True,
+   permission_classes=[permissions.AllowAny],
+)
```
Comment (fiedosiukr) [inline backend/sky_lectures/urls.py]: I guess there is no need to add permission_classes here due to the fact that swagger should be only active during the development.

**Review by fiedosiukr — CHANGES_REQUESTED** [review]

File: backend/sky_lectures/urls.py (hunk 1 above)
Comment (jakub-figat) [reply backend/sky_lectures/urls.py]: +1

File: backend/sky_lectures/urls.py (hunk 2 above)
Comment (jakub-figat) [reply backend/sky_lectures/urls.py]: If you copy stuff, at least make it appropriate for your use case

File: backend/sky_lectures/urls.py [hunk 5]
```diff
@@ -15,7 +15,25 @@
...
+
+if (sky_lectures.settings.base.DEBUG == True):
+    urlpatterns = [
+        path('admin/', admin.site.urls),
+        path('swagger/', schema_view.with_ui('swagger', cache_timeout=0), name='schema-swagger-ui')
+    ]
+else:
+    urlpatterns = [
+        path('admin/', admin.site.urls),
```
Comment (jakub-figat) [inline backend/sky_lectures/urls.py]: No need to wrap simple if statement with parentheses. Also, equality check with `True` is redundant you could go with `if sky_lectures.settings.base.DEBUG`
Moreover, `else` statement is unnecessary, and the admin pattern is duplicated

```suggestion
urlpatterns = [
    path("admin/", admin.site.urls),
]

if sky_lectures.settings.base.DEBUG:
    urlpatterns += [
        path(
            "swagger/",
            schema_view.with_ui("swagger", cache_timeout=0),
            name="schema-swagger-ui",
        )
    ]
```

File: backend/sky_lectures/urls.py [hunk 6]
```diff
@@ -15,7 +15,25 @@
 """
 from django.contrib import admin
 from django.urls import path
+from drf_yasg import openapi
+from drf_yasg.views import get_schema_view
+import sky_lectures.settings.base
```
Comment (jakub-figat) [inline backend/sky_lectures/urls.py]: You should get settings like this:
```suggestion
from django.conf import settings
```

https://docs.djangoproject.com/en/3.2/topics/settings/#using-settings-in-python-code

**Review by fiedosiukr — APPROVED** [review]

**Pull request merged**